unity/Assets/Scripts/TileSystem/Tile.cs

```diff
@@ -157,5 +168,24 @@ public void RequestRemoval()
         {
             RemovalRequestedEvent?.Invoke(this.Coordinate);
         }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Tile tile && tile.Behaviour != null && tile.Nature != null && tile.Type != null)
+            {
+                return this.Behaviour.Behaviour.Equals(tile.Behaviour.Behaviour)
+                    && this.Nature.Nature.Equals(tile.Nature.Nature)
+                    && this.Type.Type.Equals(tile.Type.Type);
+            }
+            else
+            {
+                return false;
+            }
+        }
```

Comment:
You override `Equals` but not `GetHashCode`. It is highly recommended to override either none or both at the same time.

[See here for more information](https://stackoverflow.com/a/371348)

(Even though I agree, that we probably won't run into problmes regarding this *at the moment*.)